webgui/app.profile.js

```diff
@@ -0,0 +1,111 @@
+    // This defaults to "normal" (strip all but warn and error) if not provided.
+    stripConsole: 'normal', // all | warn | normal | none
+
+    // The default selector engine is not included by default in a dojo.js build in order to make mobile builds
+    // smaller. We add it back here to avoid that extra HTTP request. There is also an "acme" selector available; if
+    // you use that, you will need to set the `selectorEngine` property in index.html, too.
+    selectorEngine: 'lite',
+
+    // Any module in an application can be converted into a "layer" module, which consists of the original module +
+    // additional dependencies built into the same file. Using layers allows applications to reduce the number of HTTP
+    // requests by combining all JavaScript into a single file.
+    layers: {
+        // This is the main loader module. It is a little special because it is treated like an AMD module even though
+        // it is actually just plain JavaScript. There is some extra magic in the build system specifically for this
+        // module ID.
+        'dojo/dojo': {
+            // By default, the build system will try to include `dojo/main` in the built `dojo/dojo` layer, which adds
+            // a bunch of stuff we do not want or need. We want the initial script load to be as small and quick to
+            // load as possible, so we configure it as a custom, bootable base.
+            boot: true,
+            customBase: true,
+
+            include: [
+                'dojo/_base/array',
+                'dojo/dom-construct',
+                'dojo/domReady'
+            ]
+        },
+
+        // Further application-specific layers. Note that when you create a new layer, the module referenced by the
+        // layer is always included in the layer, so it does not need to be explicitly defined in the `include` array.
+        'codecompass/bundle': {
+            include: [
+                'codecompass/codecompass',
+                'codecompass/view/codeBites',
+                'codecompass/view/diagram',
+                'codecompass/view/fileManager',
+                'codecompass/view/header',
+                'codecompass/view/infobox',
+                'codecompass/view/infoPage',
+                'codecompass/view/infoTree',
+                'codecompass/view/text',
+                'codecompass/view/cppDiagram',
+                'codecompass/view/cppInfoTree',
+                'codecompass/view/cppMenu',
+                'codecompass/view/cppReparseFileAST',
+                'codecompass/view/cppReparseMenu',
+                'codecompass/view/gitBlame',
+                'codecompass/view/gitCommitView',
+                'codecompass/view/gitMenu',
+                'codecompass/view/gitNavigator',
+                'codecompass/view/gitUtil',
+                'codecompass/view/metrics',
+                'codecompass/view/searchFields',
+                'codecompass/view/searchResult'
```

Comment:
I am conflicted about hardcoding the list of files of **plugins** in something that is **outside** the `plugin` directory. This change is reversing the efforts we made to make the architecture pluginable.

Plugins should work in a way that if I have, let's say, a Lua plugin, I just throw it into `plugins/` and build CodeCompass with it present, and it works...
This is an additional configuration step that complicates matters...

I know about #124 still not done, but we shouldn't *introduce* a similar problem.

CMake could be able to generate this particular file at build time from a template, putting the concrete list of files in from glob expressions.

Reply:
Yes, I agree. I will try to introduce some CMake magic here :)

Reply:
Done it based on the `install_webplugin` function.